src/Build/Evaluation/Expander.cs

```diff
@@ -2786,27 +2837,27 @@ internal static IEnumerable<KeyValuePair<string, S>> ExecuteStringFunction(
                         // We pass in the existing item so we can copy over its metadata
                         if (include.Length > 0)
                         {
-                            yield return new KeyValuePair<string, S>(include, item.Value);
+                            transformedItems.Add(new KeyValuePair<string, S>(include, item.Value));
                         }
                         else if (includeNullEntries)
                         {
-                            yield return new KeyValuePair<string, S>(null, item.Value);
+                            transformedItems.Add(new KeyValuePair<string, S>(null, item.Value));
                         }
                     }
                 }
 
                 /// <summary>
                 /// Intrinsic function that returns the items from itemsOfType with their metadata cleared, i.e. only the itemspec is retained.
```

Comment:
nit: saying "returns" is perhaps a bit misleading (applies to all the intrinsic functions)